main.php

```diff
@@ -47,7 +47,13 @@
             <header id="writr__masthead" class="site-header" role="banner">
                 <?php
                     $logoSize = array();
-                    $logo = tpl_getMediaFile(array(':wiki:logo.png', ':logo.png', 'images/logo.png'), false, $logoSize);
+                    $logoImages = array();
+                    if(tpl_getConf('doLogoChangesByNamespace')){
+                        $logoImages[] = getNS($ID).':logo.png';
```

Comment:
I have put that one (:wiki:logo.png) back. Other then that, I was trying to add support for namespace logos. The only way I saw was by using getNS to point to the current namespace. This however, now that I am thinking about it, doesn't check by hierarchy. Only the current one. So I'll need to make additionnal changes if I want it to work as intended.

Reply:
This was a feature request that I wanted to implement as I could easily see the use case. So I thought why not. But before completing this, I'll have to read a little bit more about the `tpl_getMediaFile` function. That is to make sure I am using it correctly. Plus I need to review the array provided so that it correctly and recursively add the namespaces.

[EDIT]
Actually maybe you could help me by pointing me in the right direction. I am looking into the repository Dokuwiki/Dokuwiki, I am having difficulties finding the function in question. I found this old commit : [27833958](https://github.com/dokuwiki/dokuwiki/commit/27833958#diff-807d196eca72532cefeba4dfad525c05a8d1082a725ccf68d13bf0e61611ba2b) from 2012. In feed.php. But it seems that file has been updated a lot since then.

[EDIT]
Nevermind, just found [API documentation](https://codesearch.dokuwiki.org/xref/dokuwiki/inc/template.php). Was able to add the necessary code to enable support of namespace logos